flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecTableSourceScan.java

```diff
@@ -53,45 +52,41 @@ public class BatchExecTableSourceScan extends CommonExecTableSourceScan
         implements BatchExecNode<RowData> {
 
     // Avoids creating different ids if translated multiple times
-    private final String dynamicFilteringDataListenerID = UUID.randomUUID().toString();
+    @Nullable private final String dynamicFilteringDataListenerID;
```

Comment:
If we don't make this change, it also can work, so I think we can keep original code here, I don't recommend exposing the dynamicFilteringDataListenerID to the developer.

Reply:
Fixed